GuessTheNumber.java

```diff
@@ -20,5 +21,104 @@
+		while (targetNumberSize < 2 || targetNumberSize > 10) {
+			System.out.println("You can only enter value from 2 to 10. Please try again.");
+			targetNumberSize = input.nextInt();
+		}
+
+		int[] userGuessArray = new int[targetNumberSize];
+		int[] targetNumber = new int[targetNumberSize];
+		for (int i = 0; i < targetNumberSize; i++) {
+			targetNumber[i] = 10;
+		}
+
+		// Generate target number
+		for (int i = 0; i < targetNumberSize; i++) {
+			int digitToAdd = (int) (Math.random() * 10);
+			for (int j = 0; j < targetNumberSize; j++) {
+				if (digitToAdd == targetNumber[j] || (digitToAdd == 0 && j == 0)) {
+					digitToAdd = (int) (Math.random() * 10);
+					j = -1;
+				}
+			}
+			targetNumber[i] = digitToAdd;
+		}
+
+		// Start of game loop
+		boolean guessed = false;
+		int[] trials = new int[100];
+		int[] trialsCorrectPlaces = new int[100];
+		int[] trialsCorrectNumbers = new int[100];
+		int trialIndex = 0;
+
+		while (!guessed) {
+			if (trialIndex != 0) {
+				System.out.println("Before new guess, here are your previous trials: ");
+				for (int i = 0; i < trialIndex; i++) {
+					System.out.print("Trial number " + (i + 1) + ": " + trials[i] + ". ");
+					System.out.println(trialsCorrectNumbers[i] + " correct numbers, " + trialsCorrectPlaces[i] + " of them at the right place!");
+				}
+			}
+
+			// Get user guess and validate its size
+			int inputGuessSize = 0;
+			int inputGuess;
+			int userGuess = 0;
+
+			while (inputGuessSize != targetNumberSize) {
+				System.out.print("Enter your guess (" + targetNumberSize + " numbers): ");
+				inputGuess = input.nextInt();
+				userGuess = inputGuess;
+
+				while (inputGuess > 0) {
+					inputGuessSize++;
+					inputGuess /= 10;
+				}
+				if (inputGuessSize != targetNumberSize) {
+					System.out.println("Your guess should consist of " + targetNumberSize + " digits! Please try again.");
+					inputGuessSize = 0;
+				}
+			}
+			trials[trialIndex] = userGuess;
+
+			// Convert user input into array
+			int j = targetNumber.length - 1;
+			while (j > -1) {
+				userGuessArray[j] = userGuess % 10;
+				userGuess /= 10;
+				j--;
+			}
+
+			// Check user guess
+			int correctPlace = 0;
+			int correctNumber = 0;
+			for (int i = 0; i < targetNumber.length; i++) {
+				if (targetNumber[i] == userGuessArray[i]) {
+					correctPlace++;
+					trialsCorrectPlaces[trialIndex] = correctPlace;
+				}
+				for (int num : targetNumber) {
+					if (num == userGuessArray[i]) {
+						correctNumber++;
+						trialsCorrectNumbers[trialIndex] = correctNumber;
+					}
+				}
+			}
+			if (correctPlace == targetNumberSize) {
+				System.out.println("This is correct number! Congratulations! It took you " + (trialIndex + 1) + " trials to guess!");
+				guessed = true;
+			} else {
+				System.out.println("You have guessed " + correctNumber + " numbers, " + correctPlace + " of them at the right place!");
+				trialIndex++;
+			}
+		} // End of game loop
+	}
 }

```

Comment:
Great job
Clean code
Because the code is too long (which is OK because we haven't learned to use methods) please add some more comments